pypck/module.py

```diff
@@ -305,6 +305,120 @@ def oem_text(self) -> List[str]:
         # return ''.join([block for block in self._oem_text if block])
 
 
+class GroupMembershipRequestHandler:
+    """Request handler to request static and dynamic group membership of a module."""
+
+    def __init__(
+        self,
+        addr_conn: "ModuleConnection",
+        num_tries: int = 3,
+        timeout_msec: int = 1500,
+    ):
+        """Initialize class instance."""
+        self.addr_conn = addr_conn
+
+        self._static_groups: List[LcnAddr] = []
+        self._dynamic_groups: List[LcnAddr] = []
+
+        self.static_groups_trh = TimeoutRetryHandler(num_tries, timeout_msec)
+        self.static_groups_trh.set_timeout_callback(self.timeout_static_groups)
+        self.dynamic_groups_trh = TimeoutRetryHandler(num_tries, timeout_msec)
+        self.dynamic_groups_trh.set_timeout_callback(self.timeout_dynamic_groups)
+
+        # callback
+        addr_conn.register_for_inputs(self.process_input)
+
+        # events
+        self.static_groups_known = asyncio.Event()
+        self.dynamic_groups_known = asyncio.Event()
+
+    def process_input(self, inp: inputs.Input) -> None:
+        """Create a task to process the input object concurrently."""
+        asyncio.create_task(self.async_process_input(inp))
+
+    async def async_process_input(self, inp: inputs.Input) -> None:
+        """Process incoming input object.
+
+        Method to handle incoming commands for this specific request handler.
+        """
+        if isinstance(inp, inputs.ModStatusGroups):
+            if inp.dynamic:
+                self._dynamic_groups = inp.groups
```

Comment:
The `self.dynamic_groups_known_event.set()` must be invoked here.
Otherwise we repeat requesting group membership again and again.